lightning/src/ln/interactivetxs.rs

```diff
@@ -97,54 +97,190 @@ pub enum AbortReason {
 	InvalidTx,
 }
 
-#[derive(Debug)]
-pub struct TxInputWithPrevOutput {
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct InteractiveTxInput {
+	serial_id: SerialId,
 	input: TxIn,
 	prev_output: TxOut,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct InteractiveTxOutput {
+	serial_id: SerialId,
+	tx_out: TxOut,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct ConstructedTransaction {
+	holder_is_initiator: bool,
```

Comment:
Yeah later for applying witnesses to the correct inputs. They'll be in the order of serial_id when we get them from our counterparty and since we combined them, we'll still need to know which parity is for which party.